src/item_menu_icons.c

```diff
@@ -0,0 +1,676 @@
+    {
+        StartSpriteAffineAnim(sprite, 0);
+        sprite->callback = SpriteCallbackDummy;
+    }
+}
+
+void AddSwitchPocketRotatingBallSprite(s16 rotationDirection)
+{
+    u8 *spriteId = &gUnknown_0203CE54->unk804[1];
+    LoadSpriteSheet(&gUnknown_0857FBA0);
+    LoadSpritePalette(&gUnknown_0857FBA8);
+    *spriteId = CreateSprite(&gUnknown_0857FBB0, 16, 16, 0);
+    gSprites[*spriteId].data[0] = rotationDirection;
+}
+
+static void update_switch_pocket_rotating_ball_coords(struct Sprite *sprite)
+{
+    sprite->centerToCornerVecX = sprite->data[1] - ((sprite->data[3] + 1) & 1);
+    sprite->centerToCornerVecY = sprite->data[1] - ((sprite->data[3] + 1) & 1);
+}
+
+static void SpriteCB_SwitchPocketRotatingBallInit(struct Sprite *sprite)
+{
+    sprite->oam.affineMode = 1;
+    if (sprite->data[0] == -1)
+        sprite->affineAnims = sSpriteAffineAnimTable_857FB98;
+    else
+        sprite->affineAnims = sSpriteAffineAnimTable_857FB9C;
+
+    InitSpriteAffineAnim(sprite);
+    sprite->data[1] = sprite->centerToCornerVecX;
+    sprite->data[1] = sprite->centerToCornerVecY;
+    update_switch_pocket_rotating_ball_coords(sprite);
+    sprite->callback = SpriteCB_SwitchPocketRotatingBallContinue;
+}
+
+static void SpriteCB_SwitchPocketRotatingBallContinue(struct Sprite *sprite)
+{
+    sprite->data[3]++;
+    update_switch_pocket_rotating_ball_coords(sprite);
+    if (sprite->data[3] == 16)
+        RemoveBagSprite(1);
+}
+
+void AddBagItemIconSprite(u16 itemId, u8 id)
+{
+    u8 *spriteId = &gUnknown_0203CE54->unk804[id + 2];
+    if (*spriteId == 0xFF)
+    {
+        u8 iconSpriteId;
+
+        FreeSpriteTilesByTag(id + 102);
+        FreeSpritePaletteByTag(id + 102);
+        iconSpriteId = AddItemIconSprite(id + 102, id + 102, itemId);
+        if (iconSpriteId != MAX_SPRITES)
+        {
+            *spriteId = iconSpriteId;
+            gSprites[iconSpriteId].pos2.x = 24;
+            gSprites[iconSpriteId].pos2.y = 88;
+        }
+    }
+}
+
+void RemoveBagItemIconSprite(u8 id)
+{
+    RemoveBagSprite(id + 2);
+}
+
+void sub_80D4FAC(void)
+{
+    sub_8122344(&gUnknown_0203CE54->unk804[4], 8);
+}
+
+void sub_80D4FC8(u8 arg0)
+{
+    sub_81223FC(&gUnknown_0203CE54->unk804[4], 8, arg0);
+}
+
+void sub_80D4FEC(u8 arg0)
+{
+    sub_8122448(&gUnknown_0203CE54->unk804[4], 136, 120, (arg0 + 1) * 16);
+}
+
+static void sub_80D5018(void *mem0, void *mem1)
+{
+    u8 i, j;
+
+    memset(mem1, 0, 0x800);
+    mem1 += 0x100;
+    for (i = 0; i < 6; i++)
+    {
+        mem1 += 0x20;
+        for (j = 0; j < 6; j++)
+        {
+            memcpy(mem1, mem0, 0x20);
+            mem1 += 0x20;
+            mem0 += 0x20;
+        }
+        if (i != 5)
+            mem1 += 0x20;
+    }
+}
+
+static void sub_80D5070(u8 berryId)
+{
+    struct CompressedSpritePalette pal;
+
+    if (berryId == ITEM_TO_BERRY(ITEM_ENIGMA_BERRY) - 1)
+        IsEnigmaBerryValid(); // what's the point of calling it if the return value is ignored?
```

Comment:
Try and see if an empty if statement block works here, as it may have gotten optimized out